cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -170,13 +172,21 @@ func (m *AwsManager) GetAsgForInstance(instance *AwsRef) (*Asg, error) {
 	if config, found := m.asgCache[*instance]; found {
 		return config, nil
 	}
+	if _, found := m.instancesNotInManagedAsg[*instance]; found {
+		// The instance is already known to not belong to any configured ASG
+		// Skip regenerateCache so that we won't unnecessarily call DescribeAutoScalingGroups
+		// See https://github.com/kubernetes/contrib/issues/2541
+		return nil, nil
+	}
 	if err := m.regenerateCache(); err != nil {
 		return nil, fmt.Errorf("Error while looking for ASG for instance %+v, error: %v", *instance, err)
 	}
 	if config, found := m.asgCache[*instance]; found {
 		return config, nil
 	}
 	// instance does not belong to any configured ASG
+	glog.V(6).Infof("Instance %+v is not in any ASG managed by CA. CA is now memorizing the fact not to unnecessarily call AWS API afterwards trying to find the unexistent managed ASG for the instance", *instance)
+	m.instancesNotInManagedAsg[*instance] = struct{}{}
```

Comment:
Are there possible race conditions here? i.e. could a recently launched instance be added to this list because it was missing tags for short period of time?